File: 2022/01.py
# https://adventofcode.com/2022/day/1

with open("01data.txt", "r") as datafile:
    lines = datafile.readlines()

elves = []
i = 0
for line in lines:
    if line == '\n':
        i += 1
    else:
        if len(elves) == i:
            elves.append(int(line))
        else:
            elves[i] += int(line)

# Part 1
print(max(elves))

# Part 2
elves.sort(reverse=True)
print(elves[0] + elves[1] + elves[2])

File: 2022/01.dart
// https://adventofcode.com/2022/day/1
// Solved in Python then reimplemented in Dart

import 'dart:io';
import 'dart:math';

void main() {
  final List<String> input = File('01data.txt').readAsLinesSync();

  List<int> elves = [];
  int elfIndex = 0;

  input.forEach((line) {
    if (line.isEmpty) {
      elfIndex++;
    } else {
      if (elves.length == elfIndex) {
        elves.add(int.parse(line));
      } else {
        elves[elfIndex] += int.parse(line);
      }
    }
  });

  // Part 1
  print(elves.reduce(max));

  // Part 2
  elves.sort();
  print(
    elves.reversed.take(3).reduce(
      (total, next) => total + next
    )
  );
}
